### PR TITLE
Fix build of substrate connect and testing

### PR DESCRIPTION
**Pull request by wirednkod**
Fix the build structure to be more flat than before by removing unnecessary files that should not be included in the built.
Replaced the ts-jest with babel-jest but eventually that was removed as well from the package.
In addition paths where needed, altered to include extension in order to support both cjs and esm.

### CODE REVIEW

File: packages/connect/package.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
   "description": "Substrate-connect to Smoldot clients. Using either substrate extension with predefined clients or an internal smoldot client based on chainSpecs provided.",
   "author": "Parity Team <admin@parity.io>",
   "license": "GPL-3.0-only",
+  "type": "module",
```
Comment (wanderer) [inline packages/connect/package.json:7]: did the position of this change for a reason?

File: packages/connect/tsconfig.test.json [hunk 2]
```diff
@@ -1,14 +1,16 @@
 {
     "compileOnSave": true,
     "compilerOptions": {
-        "target": "ES2018",
+        "target": "ESNEXT",
```
Comment (wanderer) [inline packages/connect/tsconfig.test.json:4]: why next and not 2020?

**Review by wanderer — APPROVED** [review]

File: packages/connect/package.json (hunk 1 above)
Comment (wirednkod) [reply packages/connect/package.json:7]: Absolutely no reason :)

File: packages/connect/tsconfig.test.json (hunk 2 above)
Comment (wirednkod) [reply packages/connect/tsconfig.test.json]: For now - ESNEXT and ES2020 are same - but to be on safe side I will change that to ES2020

**Pull request merged**